### DISCUSSION TITLE
AbiErrorSignatureNotFoundError error

### DISCUSSION THREAD

**Discussion by NetFreak26** [wevm/viem#1577] [Question]
I am using viem@1.2.12 and trying to estimate the gas cost:

```
                 const account = privateKeyToAccount(privateKey as Hex);
                 const walletClient = createWalletClient({
                      account: account,
                      chain: chain,
                      transport: http(rpcUrl)
                  });
                 const address = walletClient.account.address;
                 const estimatedGas =
                    await this.publicClient.estimateContractGas({
                        ...lendingMarketControllerContract,
                        account: address,
                        functionName: 'executeLiquidationCall',
                        args: [
                            collateralCcy,
                            debtCcy,
                            debtMaturity,
                            account as Hex,
                        ],
                    });
```

This code is giving below error:

AbiErrorSignatureNotFoundError: Encoded error signature "************" not found on ABI.
Make sure you are using the correct ABI and that the error exists on it.
You can look up the signature here: https://openchain.xyz/signatures?query=*********

Can someone please help?

**Comment by f1lander** [comment]
I'm having kind of the same error, did you find a solution? I trying to call a function that exists in the ABI but I got the reverted error code, "functiondoesnotexists" but it is happening randomly some time works, suing viem 2.9, or If I call another function first that does no have any error and the the one that's erroring works.

**Comment by Ian-Bright** [comment]
Same issue
